backend/api/src/main/java/com/virtuslab/gitmachete/backend/api/IGitMacheteRepositorySnapshot.java

```diff
@@ -29,7 +29,10 @@ public interface IGitMacheteRepositorySnapshot {
   Option<IExecutionResult> executeMachetePreRebaseHookIfPresent(IGitRebaseParameters gitRebaseParameters)
       throws GitMacheteException;
 
-  OngoingRepositoryOperation getOngoingRepositoryOperation();
+  interface IOngoingRepositoryOperationInfo {
```

Comment:
Doesn't make much sense to separate interface and implementation for a basic data class like that, just declaring `OngoingRepositoryOperationInfo` in `backend:api` Gradle module is okay. The boundary since when it makes is sense to separate intf&impl is somewhat fuzzy (not to say arbitrary)... but here I'd opt for simply just using concrete class even in the API module

Reply:
Or actually, it isn't so fuzzy: the question you need to ask yourself is whether keeping the impl class inside api module **would expose any underlying details about the underlying implementation in the api**? In our case, would you need to leak details about gitCore:jGit to backend:api? We strive to keep backend:api INdependent from gitCore:jGit. Or in general, we keep all api modules independent from all impl modules.

Reply:
And in case of this particular class, you won't really expose anything about backend:impl's deps (gitCore:jGit) by declaring it as concrete class in backend:api